onnx/defs/schema.cc

```diff
@@ -22,14 +22,18 @@ namespace ONNX_NAMESPACE {
 // -1 means ONNX schema hasn't been loaded yet
 // 0 means all versions of ONNX schema have been loaded
 // Other positive integer means the ONNX schemas for the specified version have been loaded
+#ifdef __ONNX_DISABLE_STATIC_REGISTRATION
```

Comment:
Do we need documentation for this flag?

Reply:
Currently it's a cmake arg [here](https://github.com/onnx/onnx/blob/main/CMakeLists.txt#L28) with a brief explanation. Where is the suitable place to add documentation?

Reply:
I don't find a top-level `.md` to add documentation on this cmake arg, apart from the description in `CMakeLists.txt`. Pls let me know where you'd like it added. Beside it looks like this cmake arg is not in `setup.py`, so I added it there in the new commit.

Let me know what else i can address. Ty!

Reply:
Thanks for making the updates. I’m not sure either. Will leave it for others to comment

Reply:
Why do we need this flag in the first place?

Reply:
@justinchuby This flag was introduced in #3266, and explained there. 
By default ONNX loads opset schema of all versions. #3266 introduced this compile-time flag to not register any schema by default, and leave the runtime to register specific version of opset schema as needed.

This current PR expands the API on that, so that (1) a runtime can now de-register schema when no longer need, and (2) a runtime will not incur error when registering multiple versions (which can encounter duplicate schema request).

The motivation is to reduce MaxRSS in runtime.